### PR TITLE
Updated linter configs

### PR DESCRIPTION
**Pull request by ammar92**
### Changes
Enabled and fixed these ruff plugins:
- `SIM`

`kat_webpage_analysis` now has a `USERAGENT` environment setting

More to enable and fix soon.

### Issue link
#784 

---

### Code Checklist
- [x] This PR only contains functionality relevant to the issue; tickets have been created for newly discovered issues.
- [ ] I have written unit tests for the changes or fixes I made.
- [ ] For any non-trivial functionality, I have added integration and/or end-to-end tests.
- [x] I have performed a self-review of my code and refactored it to the best of my abilities.

### Communication
- [ ] I have informed others of any required `.env` changes files if required and changed the `.env-dist` accordingly.
- [ ] I have made corresponding changes to the documentation, if necessary.

---
## Checklist for code reviewers:
Copy-paste the checklist from [the docs/source/templates folder](/docs/source/templates/pull_request_template_review_code.md).

---
## Checklist for QA:
Copy-paste the checklist from [the docs/source/templates folder](/docs/source/templates/pull_request_template_review_qa.md).

### CODE REVIEW

File: boefjes/boefjes/plugins/kat_binaryedge/protocols/normalize.py [hunk 1]
```diff
@@ -116,16 +116,11 @@ def run(normalizer_meta: NormalizerMeta, raw: Union[bytes, str]) -> Iterable[OOI
...
-                # FIXME: new KAT-Finding (high)?
-                pass  # The server is vulnerable and real attacks are feasible
-            elif robot == "NOT_VULNERABLE_NO_ORACLE":
-                pass  # The server supports RSA cipher suites but does not act as an oracle
-            elif robot == "NOT_VULNERABLE_RSA_NOT_SUPPORTED":
-                pass  # The server does not supports RSA cipher suites
-            elif robot == "UNKNOWN_INCONSISTENT_RESULTS":
-                # FIXME: KATFinding (low)?
-                pass  # Could not determine whether the server is vulnerable or not
+            if robot in (
```
Comment (underdarknl) [inline boefjes/boefjes/plugins/kat_binaryedge/protocols/normalize.py:119]: Lets add the original comments somewhere.

File: mula/scheduler/queues/pq.py [hunk 2]
```diff
@@ -139,24 +139,21 @@ def push(self, p_item: models.PrioritizedItem) -> Optional[models.PrioritizedIte
...
+                    self.allow_updates and item_changed and item_on_queue,
+                    self.allow_priority_updates and priority_changed and item_on_queue,
+                    not item_on_queue,
+                )
             )
 
-            priority_changed = False if not item_on_queue or p_item.priority == item_on_queue.priority else True
-
-            allowed = False
-            if item_on_queue and self.allow_replace:
```
Comment (underdarknl) [inline mula/scheduler/queues/pq.py:151]: Im not sure we should pack this code further. There's a reason some of this is written out like this.

Comment (ammar92) [reply mula/scheduler/queues/pq.py:151]: I think the linter in general did actually do a good job here. We could ignore specific statements for the linter, but in general, the SIM option improves and simplifies such expressions.

Rework: In this case, I managed to improve these expressions by applying De Morgan's law and negating the `==` operators which should be a bit easier to read now?

File: octopoes/bits/missing_dkim/missing_dkim.py [hunk 3]
```diff
@@ -17,12 +17,11 @@ def run(
...
-    ):
-        if not additional_oois:
-            ft = KATFindingType(id="KAT-NO-DKIM")
-            yield ft
-            yield Finding(
-                ooi=input_ooi.reference,
-                finding_type=ft.reference,
-                description="This hostname does not support DKIM records",
-            )
+    ) and not additional_oois:
```
Comment (dekkers) [inline octopoes/bits/missing_dkim/missing_dkim.py:20]: There isn't a reason to put the parentheses only around the first two terms, I think moving `and not additional_oois` inside makes it clearer, maybe also add a comment like `# only report finding when there is no DKIM record`. The same for DMARC and SPF bits.

File: octopoes/octopoes/repositories/ooi_repository.py [hunk 4]
```diff
@@ -503,10 +504,8 @@ def list_neighbours(self, references: Set[Reference], paths: Set[Path], valid_ti
     def save(self, ooi: OOI, valid_time: datetime, end_valid_time: Optional[datetime] = None) -> None:
         # retrieve old ooi
         old_ooi = None
-        try:
+        with contextlib.suppress(ObjectNotFoundException):
             old_ooi = self.get(ooi.reference, valid_time=valid_time)
-        except ObjectNotFoundException:
-            pass
```
Comment (dekkers) [inline octopoes/octopoes/repositories/ooi_repository.py:509]: I think moving  `old_ooi = None` to the exception handler makes the intention of the code clearer than using `contextlib.suppress`. What the code does is set `old_ooi` to the value returned by `self.get` and set it to `None` when `ObjectNotFoundException` is thrown. The same for the save in the other two repository classes.

**Review by dekkers — CHANGES_REQUESTED** [review]

File: octopoes/bits/missing_dkim/missing_dkim.py (hunk 3 above)
Comment (ammar92) [reply octopoes/bits/missing_dkim/missing_dkim.py:20]: The parentheses are a format thing because of multiline stuff. Good point about the comment

**Review by dekkers — APPROVED** [review]

**Pull request merged**